### PR TITLE
Mkdocs-based documentation website

### PR DESCRIPTION
**Pull request by jponge**

### CODE REVIEW

File: docs/api-translation.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# API translation
+
+The Vert.x Mutiny bindings are generated from existing APIs.
```
Comment (cescoffier) [inline docs/api-translation.md:3]:
```suggestion
The Vert.x Mutiny bindings are generated from the existing Vert.x APIs.
```

File: docs/api-translation.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+## Vert.x and asynchronous operation methods
+
+Vert.x is a toolkit that simplifies asynchronous I/O programming.
+The Vert.x APIs rely on simple conventions to denote methods doing asynchronous operations, as in:
+
+```java
+public void someOperation(Foo a, Bar b, Baz c, Handler<AsyncResult<T>> handler) {
```
Comment (cescoffier) [inline docs/api-translation.md:11]: In the last code generator, this switch to considering the future version.

File: docs/api-translation.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    // (...)
+}
+```
+
+Here `someOperation` is a method where `a`, `b` and `c` are _parameters_, and `handler` is a _callback_:
+
+* the callback is notified when the asynchronous operation done in `someOperation` completes, and
```
Comment (cescoffier) [inline docs/api-translation.md:18]:
```suggestion
* the callback is notified when the asynchronous operation done in `someOperation` completes (_continuation_), and
```

File: docs/api-translation.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+!!! note "Method erasure"
+    The Mutiny code generator _erases_ methods.
+    Given an asynchronous method `foo(Handler<AsyncResult<T>>)`, then it is not present in the generated shim as it is replaced with `Uni<T> foo()`.
+
+## Helper methods
+
+Given an asynchronous operation method, the Mutiny code generator also provides 2 further variants:
+
+* `xAndAwait()` to block the caller thread until the outcome is received, or throw a `RuntimeException` when a failure arises, and
+* `xAndForget()` to trigger the operation then discard the outcome.
```
Comment (cescoffier) [inline docs/api-translation.md:66]: Maybe mentioned that the failure is logged in this case, but you can't react to it.

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
+# Overview
+
+[Eclipse Vert.x](https://vertx.io/) is a popular toolkit for writing reactive applications on the JVM.
```
Comment (cescoffier) [inline docs/index.md:3]:
```suggestion
[Eclipse Vert.x](https://vertx.io/) is the leading toolkit for writing reactive applications on the JVM.
```

**Review by cescoffier — CHANGES_REQUESTED** [review]
Looks great! What's holding you to publish this?

**Comment by jponge** [comment]
> Looks great! What's holding you to publish this?

I'm setting up CI 😉 

I could not make it all work locally with `act`, so I'll have to merge then test by pushing.

File: .build/render-documentation.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+set -euo pipefail
+IFS=$'\n\t'
+
+mvn -f vertx-mutiny-clients/pom.xml pre-site
+cp -R vertx-mutiny-clients/target/site/apidocs docs/
+
+PROJECT_VERSION=$(cat .github/project.yml | yq eval '.release.current-version' -)
+mike deploy --push --force --update-aliases $PROJECT_VERSION latest
```
Comment (cescoffier) [inline .build/render-documentation.sh:9]: you're naming your tools now? :-D

**Review by cescoffier — APPROVED** [review]

File: .build/render-documentation.sh (hunk 6 above)
Comment (jponge) [reply .build/render-documentation.sh:9]: Well `mike` :-D

**Pull request merged**